services/sysinfo/client/client.go

```diff
@@ -0,0 +1,105 @@
+*/
+
+// Package client provides the client interface for 'sysinfo'
+package client
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/google/subcommands"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/Snowflake-Labs/sansshell/client"
+	pb "github.com/Snowflake-Labs/sansshell/services/sysinfo"
+	"github.com/Snowflake-Labs/sansshell/services/util"
+)
+
+const subPackage = "sysinfo"
+
+func init() {
+	subcommands.Register(&sysinfoCmd{}, subPackage)
+}
+
+func (*sysinfoCmd) GetSubpackage(f *flag.FlagSet) *subcommands.Commander {
+	c := client.SetupSubpackage(subPackage, f)
+	c.Register(&uptimeCmd{}, "")
+	return c
+}
+
+type sysinfoCmd struct{}
+
+func (*sysinfoCmd) Name() string { return subPackage }
+func (p *sysinfoCmd) Synopsis() string {
+	return client.GenerateSynopsis(p.GetSubpackage(flag.NewFlagSet("", flag.ContinueOnError)), 2)
+}
+func (p *sysinfoCmd) Usage() string {
+	return client.GenerateUsage(subPackage, p.Synopsis())
+}
+func (*sysinfoCmd) SetFlags(f *flag.FlagSet) {}
+
+func (p *sysinfoCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	c := p.GetSubpackage(f)
+	return c.Execute(ctx, args...)
+}
+
+type uptimeCmd struct{}
+
+func (*uptimeCmd) Name() string     { return "uptime" }
+func (*uptimeCmd) Synopsis() string { return "Get the uptime of the system" }
+func (*uptimeCmd) Usage() string {
+	return `uptime :
+	 Print the uptime of the system in below format: System_idx (ip:port) up for X days, X hours, X minutes, X seconds | total X seconds (X is a placeholder that will be replaced)
+`
+}
+
+func (p *uptimeCmd) SetFlags(f *flag.FlagSet) {}
+
+func (p *uptimeCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	state := args[0].(*util.ExecuteState)
+	c := pb.NewSysInfoClientProxy(state.Conn)
+
+	resp, err := c.UptimeOneMany(ctx, &emptypb.Empty{})
+	if err != nil {
+		// Emit this to every error file as it's not specific to a given target.
+		for _, e := range state.Err {
+			fmt.Fprintf(e, "All targets - could not info servers: %v\n", err)
+		}
+		return subcommands.ExitFailure
+	}
+
+	retCode := subcommands.ExitSuccess
+	for r := range resp {
+		if r.Error != nil {
+			fmt.Fprintf(state.Err[r.Index], "info for target %s (%d) returned error: %v\n", r.Target, r.Index, r.Error)
+			// If any target had errors it needs to be reported for that target but we still
+			// need to process responses off the channel. Final return code though should
+			// indicate something failed.
+			retCode = subcommands.ExitFailure
+			continue
+		}
+		d := r.Resp.UptimeSeconds.AsDuration()
+		hours := int(d.Hours())
+		minutes := int(d.Minutes()) % 60
+		seconds := int(d.Seconds()) % 60
+		days := hours / 24
+		hours = hours % 24
+		fmt.Fprintf(state.Out[r.Index], "System_%d (%s) up for %d days, %d hours, %d minutes, %d seconds | total %.f seconds \n", r.Index, r.Target, days, hours, minutes, seconds, d.Seconds())
```

Comment:
Is there value in the System_ part of this?

It might be better to be consistent with https://github.com/Snowflake-Labs/sansshell/blob/main/services/healthcheck/client/client.go#L89 and print out a line like

```
10.0.0.1 (0) up for 1 days, 2 hours, 3 minutes, 4 seconds | total 10 seconds
```

Reply:
This is what the output looks like:
`$ cd cmd/sanssh && go build && ./sanssh --targets=localhost sysinfo uptime`
`System_0 (localhost:50042) up for 0 days, 19 hours, 34 minutes, 36 seconds | total 70476 seconds `
the number after System_ is the index of the hosts the sanssh client is connecting to.  I think both work fore me, but considering consistency, I will go with your suggestion.